integration-tests/src/integrationTest/org/apache/cassandra/sidecar/lifecycle/LifecycleProviderIntegrationTester.java

```diff
@@ -0,0 +1,211 @@
+                                        "STOPPED", "RUNNING", "CONVERGING", "Submitting start task for instance");
+        waitForLastUpdateToConverge("Instance has started", TIMEOUT_SECONDS);
+        waitForCqlStatus("OK", TIMEOUT_SECONDS);
+
+        // Confirm that the node is up again with GET /lifecycle
+        HttpResponse<Buffer> lifecycleInfoAfterRecovery = getLifecycle();
+        assertExpectedLifecycleResponse(lifecycleInfoAfterRecovery, OK.code(),
+                                        "RUNNING", "RUNNING", "CONVERGED", "Instance has started");
+
+        // Stop node via PUT /lifecycle
+        HttpResponse<Buffer> lifecycleUpdateStop = putLifecycle("stop");
+        assertExpectedLifecycleResponse(lifecycleUpdateStop, ACCEPTED.code(),
+                                        "RUNNING", "STOPPED", "CONVERGING", "Submitting stop task for instance");
+
+        // Allow time for the async stop task to complete
+        waitForLastUpdateToConverge("Instance has stopped", TIMEOUT_SECONDS);
+
+        // GET /lifecycle after node has stopped
+        HttpResponse<Buffer> lifecycleInfoAfterStop = getLifecycle();
+        assertExpectedLifecycleResponse(lifecycleInfoAfterStop, OK.code(),
+                                        "STOPPED", "STOPPED", "CONVERGED", "Instance has stopped");
+
+        // CQL status should be down since instance is stopped
+        assertThat(getCqlStatus()).isEqualTo("NOT_OK");
+
+        // Try stopping node with PUT /lifecycle again, should be idempotent
+        HttpResponse<Buffer> lifecycleInfoOnStopAgain = putLifecycle("stop");
+        assertExpectedLifecycleResponse(lifecycleInfoOnStopAgain, OK.code(),
+                                        "STOPPED", "STOPPED", "CONVERGED", "Instance has stopped");
+    }
+
+    private void assertExpectedLifecycleResponse(
+    HttpResponse<Buffer> response,
+    int expectedStatusCode,
+    String currentState,
+    String desiredState,
+    String status,
+    String lastUpdate)
+    {
+        assertThat(response.statusCode()).isEqualTo(expectedStatusCode);
+        JsonObject body = response.bodyAsJsonObject();
+        assertThat(body.getString("current_state")).isEqualTo(currentState);
+        assertThat(body.getString("desired_state")).isEqualTo(desiredState);
+        assertThat(body.getString("status")).isEqualTo(status);
+        assertThat(body.getString("last_update")).isEqualTo(lastUpdate);
+    }
+
+    private HttpResponse<Buffer> getLifecycle()
+    {
+        return getBlocking(client
+                           .get(sidecarPort, sidecarHost, "/api/v1/cassandra/lifecycle")
+                           .send());
+    }
+
+    private HttpResponse<Buffer> putLifecycle(String desiredState)
+    {
+        return getBlocking(client
+                           .put(sidecarPort, sidecarHost, "/api/v1/cassandra/lifecycle")
+                           .sendBuffer(Buffer.buffer("{\"state\":\"" + desiredState + "\"}")));
```

Comment:
NIT: this is a little bit easier to read and a bit simpler to build using `io.vertx.core.json.JsonObject` 
```suggestion
                           .sendBuffer(JsonObject.of("state", desiredState).toBuffer()));
```